holidays/countries/thailand.py

```diff
@@ -12,156 +12,880 @@
-        add_holiday(self.cnls.vesak_may_date(year), "Buddha's Birthday")
-
-        # Coronation Day.
-        if year <= 2016:
-            add_holiday(date(year, MAY, 5), "Coronation Day")
+        ########################
+        #
+        # FIXED DATED HOLIDAYS
+        #
+        ########################
+
+        # !!! New Year's Day !!!
+        # วันขึ้นปีใหม่
+        # Status: In-Use
+        # Starts in the present form in 1941 (B.E. 2484)
+        # Sources:
+        #   (wikisource.org 's wbm) http://tiny.cc/wa_wiki_thai_newyear_2483
+        new_years_day_en = "New Year's Day"
+
+        if year >= 1941:
+            add_holiday(date(year, JAN, 1), new_years_day_en)
+
+        # !!! New Year's Eve (in lieu) !!!
+        # วันหยุดชดเชยวันสิ้นปี
+        # Status: In-Use
+        # Added separately from New Year's Eve itself so that it would't
+        #   go over the next year
+        # Sources:
+        #   https://github.com/dr-prodigy/python-holidays/pull/929
+        #   https://resolution.soc.go.th/?prep_id=205799
+        #   https://resolution.soc.go.th/?prep_id=210744
+        new_years_eve_in_lieu_en = "New Year's Eve (in lieu)"
+
+        # Applied Automatically for Monday if on Weekends: 1961-1973
+        #  **NOTE: No New Year's Eve (in lieu) for this period
+        # No In Lieu days available: 1974-1988
+        # Case-by-Case application for Workday if on Weekends: 1989-1994
+        # Applied Automatically for Workday if on Weekends: 1995-1997
+        # Case-by-Case application for Workday if on Weekends: 1998-2000
+        # Applied Automatically for Workday if on Weekends: 2001-Present
+        if self.observed and (1995 <= year <= 1997 or year >= 2001):
+            # CASE 1: SAT-SUN -> 1 in-lieu on TUE
+            if date(year - 1, DEC, 31).weekday() == SAT:
+                self[date(year, JAN, 3)] = new_years_eve_in_lieu_en
+            # CASE 2: SUN-MON -> 1 in-lieu on TUE
+            elif date(year - 1, DEC, 31).weekday() == SUN:
+                self[date(year, JAN, 2)] = new_years_eve_in_lieu_en
+
+        # !!! Chakri Memorial Day !!!
+        # วันจักรี
+        # Status: In-Use
+        # Starts in present form in 1918 (B.E. 2461)
+        # Sources:
+        #   (ocac.got.th 's wbm) http://tiny.cc/wa_ocac_chakri
+        chakri_memorial_en = "Chakri Memorial Day"
+
+        if year >= 1918:
+            add_holiday(date(year, APR, 6), chakri_memorial_en)
+
+        # !!! Songkran Festival !!!
+        # วันสงกรานต์
+        # Status: In-Use
+        # Used to be April 1st as Thai New Year Day
+        # Initially abandoned in 1941 (B.E. 2484), declared again as
+        #   public holidays in 1948 (2491 B.E)
+        # This has its own in lieu trigger
+        # Sources:
+        #   (museumsiam.org 's wbm) http://tiny.cc/wa_museumsiam_songkran
+        #   https://resolution.soc.go.th/?prep_id=123659
+        songkran_festival_en = "Songkran Festival"
+
+        if self.observed:
```

Comment:
`observed` check - it's for "in lieu" only

Reply:
Oof, will get a fix for this in real quick